src/components/engine/engine-instances-table.tsx

```diff
@@ -276,46 +286,34 @@ const EditModal = ({
   refetch: () => void;
 }) => {
   const toast = useToast();
+  const { mutate: editInstance } = useEngineEditInstance();
   const { onClose } = disclosure;
 
-  const form = useForm({
+  const form = useForm<EditEngineInstanceInput>({
     defaultValues: {
-      name: instance?.name,
-      url: instance?.url,
+      instanceId: instance.id,
+      name: instance.name,
+      url: instance.url,
     },
   });
 
-  const onSubmit = async (data: { name: string; url: string }) => {
-    // Instance URLs should end with a /.
-    const url = data.url.endsWith("/") ? data.url : `${data.url}/`;
-
-    try {
-      const res = await fetch(`${THIRDWEB_API_HOST}/v1/engine/${instance.id}`, {
-        method: "PUT",
-        credentials: "include",
-        headers: {
-          "Content-type": "application/json",
-        },
-        body: JSON.stringify({
-          name: data.name,
-          url,
-        }),
-      });
-      if (!res.ok) {
-        throw new Error(`Unexpected status ${res.status}`);
-      }
-      toast({
-        status: "success",
-        description: "Updated an Engine instance.",
-      });
-      refetch();
-      onClose();
-    } catch (e) {
-      toast({
-        status: "error",
-        description: "Error updating an Engine instance.",
-      });
-    }
+  const onSubmit = async (data: EditEngineInstanceInput) => {
+    await editInstance(data, {
```

Comment:
no need to `await` this - `mutate` is sync and handles errors via the `onError` callback